local-cluster/tests/local_cluster.rs

```diff
@@ -4722,14 +4726,21 @@ fn test_duplicate_with_pruned_ancestor() {
 
         // Change last block parent to chain off of (purged) minority fork
         info!("For our node, changing parent of {last_majority_vote} to {last_minority_vote}");
-        purge_slots_with_count(&our_blockstore, last_majority_vote, 1);
```

Comment:
This previous method of purging does not keep descendant info, making the pruned fork unable to reattach properly after repair. `clear_unconfirmed_slot` does the right thing and is what we use in `dump_then_repair`